src/main/scala/mesosphere/marathon/core/heartbeat/HeartbeatActor.scala

```diff
@@ -92,28 +103,46 @@ object Heartbeat {
     /** Decorator generates a modified Reactor with enhanced functionality */
     trait Decorator extends (Reactor => Reactor)
 
+    abstract class Adapter(delegate: Reactor) extends Reactor {
+      override def onSkip(missed: Int): Unit = delegate.onSkip(missed)
+      override def onFailure(): Unit = delegate.onFailure()
+    }
+
     object Decorator {
       def apply(f: Reactor => Reactor): Decorator = new Decorator {
         override def apply(r: Reactor): Reactor = f(r)
       }
     }
 
-    /**
-      * withLogging decorates the given Reactor by logging messages prior to forwarding each callback
-      */
-    final val withLogging: Decorator = Decorator { r =>
-      new Reactor {
-        def onSkip(): Unit = {
-          log.info("detected skipped heartbeat")
-          r.onSkip()
+    /** afterNMisses only propagates onSkip events after nMisses are exceeded */
+    def afterNMisses(nMisses: Int): Decorator = Decorator { r =>
+      new Adapter(r) {
+        override def onSkip(missed: Int): Unit = {
+          if (missed > nMisses) {
+            super.onSkip(missed)
```

Comment:
Once you've skipped the optional heartbeat (waiting on offers, etc.) you can continue to log the `detected skipped heartbeat` right?